src/util.rs

```diff
@@ -15,11 +15,11 @@ pub(crate) fn sha3_256(data: &[u8]) -> [u8; 32] {
 use crate::error::{FromBytesError, FromBytesResult};
 use group::{CurveAffine, CurveProjective, EncodedPoint};
 
-use crate::{PK_SIZE, SK_SIZE};
+use crate::{PK_SIZE, SIG_SIZE, SK_SIZE};
 use ff::{Field, PrimeField};
-use pairing::bls12_381::{Fr, FrRepr, G1Affine, G1};
+use pairing::bls12_381::{Fr, FrRepr, G1Affine, G2Affine, G1, G2};
 
-pub(crate) fn fr_from_be_bytes(bytes: [u8; SK_SIZE]) -> FromBytesResult<Fr> {
+pub fn fr_from_be_bytes(bytes: [u8; SK_SIZE]) -> FromBytesResult<Fr> {
```

Comment:
The problem with using `TryFrom` is there's no way to communicate they must use big endian byte order. If someone tries to convert little endian bytes into Fr it would/might work but give an unintended value for Fr.

`fr_from_be_bytes` makes it really clear that big endian bytes are required.

We could add a comment above the `TryFrom` impl to use big endian, but to me the most ergonomic approach is to use `fr_from_be_bytes`

What do you reckon?

Reply:
good point, yeah let's leave it.